src/Pretzel/Program.cs

```diff
@@ -1,89 +1,160 @@
+                    {
+                        var program = host.GetExport<Program>();
+                        var result = await program.Run(GlobalOptions, args);
+                        WaitForClose();
+                        return result;
+                    }
+                }
+                catch (Exception ex)
                 {
-                    program.ShowHelp();
-                    return;
+                    Tracing.Error(ex.Message);
+                    WaitForClose();
+                    return -1;
                 }
+            });
+            try
+            {
+                return await rootCommand.InvokeAsync(args);
+            }
+            catch (Exception ex)
+            {
+                Tracing.Error(ex.Message);
+                WaitForClose();
+                return -1;
+            }
+        }
+
+        internal static string[] PatchSourcePath(string[] args)
+        {
+            if (args.Length > 1 && !args.Contains("-s") && !args.Contains("--source"))
+            {
+                var firstArgument = args[1];
+
+                // take the first argument after the command
+                if (firstArgument != null && !firstArgument.StartsWith("-") && !firstArgument.StartsWith("/"))
+                {
+                    var arguments = args.ToList();
+                    var path = fileSystem.Path.IsPathRooted(firstArgument)
+                        ? firstArgument
+                        : fileSystem.Path.Combine(fileSystem.Directory.GetCurrentDirectory(), firstArgument);
 
-                program.Run();
+                    path = string.IsNullOrWhiteSpace(path)
+                        ? fileSystem.Directory.GetCurrentDirectory()
+                        : fileSystem.Path.GetFullPath(path);
+
+                    arguments[1] = "-s";
+                    arguments.Insert(2, path);
+                    return arguments.ToArray();
+                }
             }
+
+            return args;
         }
 
-        private void InitializeTrace()
+        private static void InitializeTrace(bool debug)
         {
             Tracing.SetTrace(ConsoleTrace.Write);
 
-            if (parameters.Debug)
+            if (debug)
             {
                 Tracing.SetMinimalLevel(TraceLevel.Debug);
             }
         }
 
-        private void ShowHelp()
+        private async Task<int> Run(IEnumerable<Option> globalOptions, string[] args)
         {
-            Commands.WriteHelp(parameters.Options);
-            WaitForClose();
-        }
+            try
+            {
+                foreach (var option in globalOptions)
+                {
+                    CommandCollection.RootCommand.AddOption(option);
 
-        private void Run()
-        {
-            if (Commands[parameters.CommandName] == null)
+                    foreach (var command in CommandCollection.RootCommand.OfType<Command>())
+                    {
+                        command.AddOption(option);
```

Comment:
This is done to avoid duplication. Those get mapped to [PretzelBaseCommandArguments](https://github.com/Code52/pretzel/pull/347/files#diff-ff4a8a22a60456b2e4a9dd66434ab628R38) or any `Argument` that provides any of the global options by convention. `Source` `Debug` and `Safe` by now.